backend/src/apiserver/template/tekton_template.go

```diff
@@ -294,9 +294,24 @@ func (t *Tekton) injectArchivalStep(workflow util.Workflow, artifactItemsJSON ma
 					task.TaskSpec.StepTemplate.VolumeMounts = append(task.TaskSpec.StepTemplate.VolumeMounts, loggingVolumeMounts...)
 				}
 
+				moveStep := false
+				if task.TaskSpec.Results != nil && len(task.TaskSpec.Results) > 0 {
```

Comment:
So I think pretty much all the time we will not have results in the last step. Meaning eof script will not execute all the time?

Reply:
yep, because we need to move out those results file to avoid duplication